### PR TITLE
Fixes integration test setup and propagation bug

### LINKED ISSUE #11: Test failure: brave.cassandra.ITTracing.usesExistingTraceId
When the build was updated so failsafe executed integration tests (see #10), a failure was revealed in `brave.cassandra.ITTracing.usesExistingTraceId`. I could reproduce this failure locally, so it seems like a genuine issue.

```
java.lang.AssertionError: 

Expecting:
 <[CLIENT]>
to contain only once:
 <[SERVER, CLIENT]>
but some elements were not found:
 <[SERVER]>

	at brave.cassandra.ITTracing.usesExistingTraceId(ITTracing.java:98)
```

### PR DESCRIPTION
**Pull request by codefromthecrypt**
There was a propagation bug that resulted in cassandra server not
continuing inbound traces. This was hidden by incorrect integration
test setup.

Fixes #11

### CODE REVIEW

File: cassandra/src/main/java/brave/cassandra/Tracing.java [hunk 1]
```diff
@@ -96,9 +97,9 @@ protected final UUID newSession(
 
   /** This extracts the RPC span encoded in the custom payload, or starts a new trace */
   Span spanFromPayload(Tracer tracer, @Nullable Map<String, ByteBuffer> payload) {
-    ByteBuffer b3 = payload.get("b3");
+    ByteBuffer b3 = payload != null ? payload.get("b3") : null;
```
Comment (codefromthecrypt) [inline cassandra/src/main/java/brave/cassandra/Tracing.java:100]: bug 1

File: cassandra/src/main/java/brave/cassandra/Tracing.java [hunk 2]
```diff
@@ -96,9 +97,9 @@ protected final UUID newSession(
...
     if (b3 == null) return tracer.nextSpan();
-    TraceContextOrSamplingFlags extracted = B3SingleFormat.parseB3SingleFormat(b3.asCharBuffer());
+    TraceContextOrSamplingFlags extracted = B3SingleFormat.parseB3SingleFormat(UTF_8.decode(b3));
```
Comment (codefromthecrypt) [inline cassandra/src/main/java/brave/cassandra/Tracing.java:102]: bug 2

File: pom.xml [hunk 3]
```diff
@@ -251,6 +279,20 @@
       <plugin>
         <artifactId>maven-failsafe-plugin</artifactId>
         <version>${maven-failsafe-plugin.version}</version>
+        <executions>
```
Comment (codefromthecrypt) [inline pom.xml:282]: missing setup for IT

**Review by shakuzen — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
not really sure why jenkins is failing, unless it needs a kick in order to re-read the test config.

anyway will sort that separately.

**Comment by codefromthecrypt** [comment]
test related problem sorted here https://github.com/apache/incubator-zipkin-brave-cassandra/commit/3dfa9aebef2e37a8c76e9794d670e09b0828de94

**Pull request merged**